google/cloud/bigquery/job.py

```diff
@@ -3170,6 +3175,8 @@ def result(
                 request. Non-positive values are ignored.
             max_results (Optional[int]):
                 The maximum total number of rows from this request.
+            start_index (int):
+                (Optional) The zero-based index of the starting row to read.
```

Comment:
Let's move the word "optional" to the type itself, i.e. `Optional[int]`, to be consistent with other parameters.